e2e2/test/cases/nvidia/main_test.go

```diff
@@ -73,12 +86,59 @@ func TestMain(m *testing.M) {
 			}
 			return ctx, nil
 		},
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			clientset, err := kubernetes.NewForConfig(cfg.Client().RESTConfig())
+			if err != nil {
+				return ctx, err
+			}
+			nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return ctx, err
+			}
+			if *efaEnabled {
+				err := fwext.ApplyManifests(cfg.Client().RESTConfig(), efaDevicePluginManifest)
+				if err != nil {
+					return ctx, err
+				}
+				ds := appsv1.DaemonSet{
+					ObjectMeta: metav1.ObjectMeta{Name: "efa-device-plugin-daemonset", Namespace: "kube-system"},
+				}
+				err = wait.For(fwext.NewConditionExtension(cfg.Client().Resources()).DaemonSetReady(&ds),
+					wait.WithTimeout(time.Minute*5))
+				if err != nil {
+					return ctx, err
+				}
+			}
+			if *nodeType != "" {
+				for _, v := range nodes.Items {
+					if v.Labels["node.kubernetes.io/instance-type"] == *nodeType {
+						nodeCount++
+						gpu := v.Status.Capacity["nvidia.com/gpu"]
+						gpuPerNode = int(gpu.Value())
+						efa := v.Status.Capacity["vpc.amazonaws.com/efa"]
+						efaPerNode = int(efa.Value())
+					}
+				}
+			} else {
+				log.Printf("No node type specified. Using the node type %s in the node groups.", nodes.Items[0].Labels["node.kubernetes.io/instance-type"])
+				nodeCount = len(nodes.Items)
+				gpu := nodes.Items[0].Status.Capacity["nvidia.com/gpu"]
+				gpuPerNode = int(gpu.Value())
+				efa := nodes.Items[0].Status.Capacity["vpc.amazonaws.com/efa"]
+				efaPerNode = int(efa.Value())
+			}
```

Comment:
So the idea is that our deployer doesn't have to pass any info to the test binary about what instance type it deployed, the test case will just adapt to whatever it finds in the cluster? I think that's a fine idea, but I think you need to assert here that you have a single instance type in the cluster and throw an error if there's a mix.

Also why move this to `main_test.go`? It's only relevant to the multi-node tests, right?

Reply:
> Also why move this to main_test.go? It's only relevant to the multi-node tests, right?
Yes, but I plan to modify the single node test as well, so it can make the most use of the single node.